---
title: "S5_16S_beta_div"
author: "Stephen Formel"
date: "3/21/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Description and notes

Run PERMANOVA and make PCA of 16S based on Aitchison

## Load Libraries
```{r, echo = FALSE, message = FALSE}
library(readxl)
library(plyr)
library(tidyverse)
library(phyloseq)
library(data.table)
library(reshape2)
library(car)
library(cowplot)
library(compositions)
library(vegan)
library(ggcorrplot)
library(ggforce)
library(gganimate)
```

## Set up aesthetic for plotting

```{r}

#color palette
cPAL <- c("#F0E442", "#0072B2") #yellow and blue
cPAL <- c("#D55E00", "#0072B2")
```

```{r}
SF1_aes <- aes(x = PC1,
               y = PC2,
               shape = orig_soil,
               fill = oil_added)

my.scale_aes <- list(geom_point(size = 2),
                     #geom_polygon(alpha = 0.5),
                     scale_shape_manual(values = c(21,24)),
                     scale_fill_manual(values = c(cPAL, "black", "gray")),
                     theme_bw(),
                     guides(fill = guide_legend(override.aes=list(shape=21, size = 4)),
                            shape = guide_legend(override.aes=list(size = 4))),
                     #labs(shape = "Soil",
                      #    fill = "Oil"),
                     theme(plot.title = element_text(hjust = 0.5)))


```

## Import 16S phyloseq object and clean

```{r}

#read in data
S5 <- readRDS("../../data/16S/S5_16S_pseq.rds")

taxa_names(S5) <- paste0("Seq", seq(ntaxa(S5)))

S5.noCL.noMT <- S5 %>% 
  subset_taxa( Family!= "Mitochondria" & Order!="Chloroplast")

S5 <- S5.noCL.noMT
```


### Plot and PERMANOVA (adonis) of 16S - Aitchison

By Tissue through time
```{r}

#takes about 5 min on SF macbook pro
df.all <- otu_table(S5) %>% 
  as.data.frame()

df.meta.all <- sample_data(S5) %>% 
  data.frame()

#make generic for downstream analysis
df <- df.all
df.meta <- df.meta.all

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#for PERMANOVA with all chemistry
comm.df.16S <- comm.df
```

#### Plot PCA - Aitchison
All tissues

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

ggplot(data = df.PCA,
       aes(x = PC1,
           y = PC2,
           shape = orig_soil,
           fill = tissue)) +
  my.scale_aes +
  labs(title = "PCA on 16S Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)")) +
    facet_grid(cols = vars(sampling_period),
               rows = vars(oil_added)) +
  ggforce::geom_mark_ellipse(aes(linetype = orig_soil))


#NMDS wihout Soil 3- N17 as outlier
comm.df1 <- comm.df[-132,]

C <- metaMDS(comm.df1, distance = "euclidean")

D <- as.data.frame(C$points)

ggplot(data = cbind(df.meta[-132,], D),
       aes(x = MDS1,
           y = MDS2,
           shape = orig_soil,
           fill = tissue)) +
  my.scale_aes +
  facet_grid(cols = vars(sampling_period),
               rows = vars(oil_added)) +
  ggforce::geom_mark_ellipse(aes(linetype = orig_soil))

#ggsave("S5_16S_tissue_through_time.png", width = 8, height = 6)
```

## Permanova - Aitchison

```{r}

set.seed(1)
mod1 <- adonis(comm.df ~ tissue, 
               data = df.meta, 
               permutations = 9999, 
               method = "euclidean",
               strata = df.meta$sampling_period) 
mod1

B <- vegdist(comm.df1, method = "euclidean")

anova(betadisper(B, group = df.meta$tissue[-132]))

```

Everything is significant...

## Soil - Did the inoculation work?

```{r}

S5.soil <- prune_samples(sample_data(S5)$tissue=="Soil", S5)

S5.soil <- prune_samples((sample_data(S5.soil)$plantID!="3" | sample_data(S5.soil)$sampling_period!="N17"), S5.soil)
   
         
#takes about 5 min on SF macbook pro
df.soil <- otu_table(S5.soil) %>% 
  as.data.frame()

df.meta.soil <- sample_data(S5.soil) %>% 
  data.frame()

#make generic for downstream analysis
df <- df.soil
df.meta <- df.meta.soil

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#for PERMANOVA with all chemistry
comm.df.16S <- comm.df

```

#### Plot PCA - Aitchison

Just SOIL
```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

#get rid of plant 3 at N17, crazy outlier

ggplot(data = df.PCA %>% 
         filter(!(plantID=="3" & sampling_period=="N17")),
       aes(x = PC1,
           y = PC2,
           fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
  ggforce::geom_mark_ellipse(aes(linetype = plant_trt),
                             alpha = 0.2) +
  scale_linetype_manual(labels = c("No Plant",
                                   "Planted"),
                        values = c("dashed", "solid")) +
  labs(title = NULL,
       caption = "Ellipses are visual guides and have no statistical meaning",
       shape = "Oil Addition",
       fill = "Inoculum",
       linetype = "Plant Treatment",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)")) +
    facet_grid(cols = vars(sampling_period),
               labeller = labeller(plant_trt = c("Y" = "Planted", "N" = "No Plant"),
                                   sampling_period = c("N16" = "6 Months", "J17" = "1 Year", "N17" = "1.5 Years", "J18" = "2 Years"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL)

ggsave("S5_16S_SOIL_through_time.png", width = 8, height = 6)

```

### NMDS

```{r}

#plantID S3-N17 is a clear outlier

A <- comm.df[rownames(comm.df)!="S3-N17",]
A <- metaMDS(comm = A, distance = "euclidean")

cbind(df.meta[rownames(df.meta)!="S3-N17",], A$points) %>% 
  ggplot(aes(x = MDS1,
             y = MDS2,
             fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
  ggforce::geom_mark_ellipse(aes(linetype = plant_trt),
                             alpha = 0.2) +
  scale_linetype_manual(labels = c("No Plant",
                                   "Planted"),
                        values = c("dashed", "solid")) +
  labs(title = NULL,
       caption = "Ellipses are visual guides and have no statistical meaning",
       shape = "Oil Addition",
       fill = "Inoculum",
       linetype = "Plant Treatment") +
    facet_grid(cols = vars(sampling_period),
               labeller = labeller(plant_trt = c("Y" = "Planted", "N" = "No Plant"),
                                   sampling_period = c("N16" = "6 Months", "J17" = "1 Year", "N17" = "1.5 Years", "J18" = "2 Years"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL) +
  geom_text(aes(label = plantID))
```
## Alternative NMDS Plot

The panels I made above were too much and there were some critiques of the PCA, so I've made something that is visually simpler and is an NMDS.  It's hard to see the yellow though, so I'm going to change the colors to something darker.

```{r}


cbind(df.meta, A$points) %>%
  group_by(plant_trt, oil_added, orig_soil, sampling_period) %>% 
  dplyr::summarise(MDS1.mean = mean(MDS1),
            MDS2.mean = mean(MDS2)) %>% 
  ggplot(aes(x = MDS1.mean,
             y = MDS2.mean,
             group = interaction(plant_trt, oil_added, orig_soil))) +
  # geom_link2(aes(alpha = sampling_period),
  #            size = 3) +
  geom_path(aes(color = orig_soil,
                linetype = fct_rev(plant_trt)),
            arrow = arrow(type = "open", 
                          length = unit(2, "mm")),
            size = 0.5,
            show.legend = TRUE) +
  geom_point(data = . %>% 
               filter(sampling_period=="N16"),
             aes(shape = oil_added,
                 fill = orig_soil),
             size = 3) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values = c(cPAL, "black", "gray")) +
  scale_color_manual(values = c(cPAL, "black", "gray")) +
  theme_bw() +
  guides(fill = guide_legend(override.aes=list(shape=22, size = 4, linetype = NULL)),
         color = guide_legend(override.aes=list(shape=22, size = 4, linetype = NULL)),
         shape = guide_legend(override.aes=list(shape=c(21,24), size = 4, linetype = NULL))) +
  labs(title = NULL,
       caption = paste0("Stress = ", round(A$stress, 2)),
       shape = "Oil Addition",
       color = "Inoculum",
       fill = "Inoculum",
       linetype = "Plant Presence",
       x = "MDS1",
       y = "MDS1") +
  theme(legend.position = "bottom" ,legend.direction = "vertical",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank())


ggsave("S5_16S_SOIL_through_time_NMDS_vector.png", width = 6, height = 5)
```

## Permanova - Aitchison

Just SOIL
```{r}

set.seed(1)
mod1 <- adonis(comm.df ~ plant_trt*orig_soil*oil_added, 
               data = df.meta, 
               permutations = 9999, 
               method = "euclidean",
               strata = df.meta$sampling_period) 
mod1

#check with dbRDA (actually RDA because it's euclidean, but the syntax was more obvious to me in dbRDA than in RDA)

mod1 <- dbrda(comm.df ~ plant_trt*orig_soil*oil_added + Condition(sampling_period), data = df.meta, 
               distance = "euclidean") 

anova(mod1, 
      by = "margin", 
      permutations = 9999)

```

Everything is significant except for the interaction of all three. But dbRDA doesn't produce a good fit.

## Roots - Did the inoculation work?

```{r}

S5.roots <- prune_samples(sample_data(S5)$tissue=="Root", S5)

#S5.roots <- prune_samples(sample_data(S5.roots)$season=="Growing", S5.roots)

#takes about 5 min on SF macbook pro
df.roots <- otu_table(S5.roots) %>% 
  as.data.frame()

df.meta.roots <- sample_data(S5.roots) %>% 
  data.frame()

#make generic for downstream analysis
df <- df.roots
df.meta <- df.meta.roots

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#for PERMANOVA with all chemistry
comm.df.16S <- comm.df

```

#### Plot PCA - Aitchison

Just ROOTS
```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

ggplot(data = df.PCA,
       aes(x = PC1,
           y = PC2,
           fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
   ggforce::geom_mark_ellipse(aes(color = orig_soil),
                              alpha = 0.2) +
  labs(title = NULL,
       caption = "Ellipses are visual guides and have no statistical meaning",
       shape = "Oil Addition",
       fill = "Inoculum",
       color = "Inoculum",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)")) +
     facet_grid(cols = vars(season),
                labeller = labeller(season = c("Growing" = "November", "Non-Growing" = "June"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL)

# A <- p + 
#   transition_states(sampling_period,
#                     transition_length = 2,
#                     state_length = 1)


ggsave("S5_16S_ROOT_through_time.png", width = 7, height = 5.5)

```
## NMDS
```{r}

A <- comm.df
A <- metaMDS(comm = A, distance = "euclidean")

cbind(df.meta, A$points) %>% 
  ggplot(aes(x = MDS1,
             y = MDS2,
             fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
  # ggforce::geom_mark_ellipse(aes(linetype = plant_trt),
  #                            alpha = 0.2) +
  # scale_linetype_manual(labels = c("No Plant",
  #                                  "Planted"),
  #                       values = c("dashed", "solid")) +
  labs(title = NULL,
       caption = paste0("Stress = ", round(A$stress, 2)),
       shape = "Oil Addition",
       fill = "Inoculum",
       linetype = "Plant Treatment") +
    facet_grid(cols = vars(season),
                 labeller = labeller(season = c("Growing" = "November", "Non-Growing" = "June"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL) +
  scale_fill_manual(values = cPAL)

ggsave("S5_16S_ROOTS_through_time_NMDS_vector.png", width = 6, height = 4)

```

## Alternative NMDS Plot

The panels I made above were too much and there were some critiques of the PCA, so I've made something that is visually simpler and is an NMDS.  It's hard to see the yellow though, so I'm going to change the colors to something darker.

```{r}


cbind(df.meta, A$points) %>%
  group_by(oil_added, orig_soil, sampling_period) %>% 
  dplyr::summarise(MDS1.mean = mean(MDS1),
            MDS2.mean = mean(MDS2)) %>% 
  ggplot(aes(x = MDS1.mean,
             y = MDS2.mean,
             group = interaction(oil_added, orig_soil))) +
  # geom_link2(aes(alpha = sampling_period),
  #            size = 3) +
  geom_path(aes(color = orig_soil),
            arrow = arrow(type = "open", 
                          length = unit(2, "mm")),
            size = 0.5,
            show.legend = TRUE) +
  geom_point(data = . %>% 
               filter(sampling_period=="N16"),
             aes(shape = oil_added,
                 fill = orig_soil),
             size = 3) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values = c(cPAL, "black", "gray")) +
  scale_color_manual(values = c(cPAL, "black", "gray")) +
  theme_bw() +
  guides(fill = guide_legend(override.aes=list(shape=22, size = 4, linetype = NULL)),
         color = guide_legend(override.aes=list(shape=22, size = 4, linetype = NULL)),
         shape = guide_legend(override.aes=list(shape=c(21,24), size = 4, linetype = NULL))) +
  labs(title = NULL,
       caption = paste0("Stress = ", round(A$stress, 2)),
       shape = "Oil Addition",
       color = "Inoculum",
       fill = "Inoculum",
       x = "MDS1",
       y = "MDS1") +
  theme(legend.position = "bottom" ,legend.direction = "vertical",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank())

#ggsave("S5_16S_ROOTS_through_time_NMDS_vector.png", width = 8, height = 6)
```
## J18 Only to compare to Duke Results

```{r}

S5.roots <- prune_samples(sample_data(S5)$tissue=="Root", S5)
S5.roots.J18 <- prune_samples(sample_data(S5.roots)$sampling_period=="J18", S5.roots)

#takes about 5 min on SF macbook pro
df.roots <- otu_table(S5.roots.J18) %>% 
  as.data.frame()

df.meta.roots <- sample_data(S5.roots.J18) %>% 
  data.frame()

#make generic for downstream analysis
df <- df.roots
df.meta <- df.meta.roots

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#for PERMANOVA with all chemistry
comm.df.16S <- comm.df

## NMDS

A <- comm.df
A <- metaMDS(comm = A, distance = "euclidean")

cbind(df.meta, A$points) %>% 
  ggplot(aes(x = MDS1,
             y = MDS2,
             fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
  # ggforce::geom_mark_ellipse(aes(linetype = plant_trt),
  #                            alpha = 0.2) +
  # scale_linetype_manual(labels = c("No Plant",
  #                                  "Planted"),
  #                       values = c("dashed", "solid")) +
  labs(title = NULL,
       caption = paste0("Stress = ", round(A$stress, 2)),
       shape = "Oil Addition",
       fill = "Inoculum",
       linetype = "Plant Treatment") +
    # facet_grid(cols = vars(sampling_period),
             # labeller = labeller(sampling_period = c("N16" = "6 Months", "J17" = "1 Year", "N17" = "1.5 Years", "J18" = "2 Years"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL)

set.seed(1)
mod1 <- adonis(comm.df ~ orig_soil*oil_added, 
               data = df.meta, 
               permutations = 9999, 
               method = "euclidean",
               strata = df.meta$sampling_period) 
mod1
```

## Permanova - Aitchison

Just ROOTS
```{r}

set.seed(1)
mod1 <- adonis(comm.df ~ orig_soil*oil_added, 
               data = df.meta, 
               permutations = 9999, 
               method = "euclidean",
               strata = df.meta$sampling_period) 
mod1

#check with dbRDA (actually RDA because it's euclidean, but the syntax was more obvious to me in dbRDA than in RDA)

mod1 <- dbrda(comm.df ~ orig_soil + oil_added + Condition(sampling_period), data = df.meta, 
               distance = "euclidean") 

anova(mod1, 
      by = "margin", 
      permutations = 9999)

```

Main effects are significant, not the interaction. dbRDA supports this.

### What taxa are responsible?

```{r}

#quick approach based off a phyloseq tutorial
#https://joey711.github.io/phyloseq/preprocess.html

S5.roots.2 <- S5.roots

transform_sample_counts(S5.roots.2, function(OTU) OTU/sum(OTU) )
#S5.roots = filter_taxa(S5.roots, function(x) sum(x > 3) > (0.2*length(x)), TRUE)

total = median(sample_sums(S5.roots.2))
standf = function(x, t=total) round(t * (x / sum(x)))
gps = transform_sample_counts(S5.roots.2, standf)

#Filter the taxa using a cutoff of 3.0 for the Coefficient of Variation
#gpsf = filter_taxa(gps, function(x) sd(x)/mean(x) > 3.0, TRUE)

#Subset the data to Proteobacteria
gpsfb = subset_taxa(gps, Phylum=="Proteobacteria")

#Now let’s summarize this slice of the data with some graphics.

plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Bacteroidetes")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Cyanobacteria")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Actinobacteria")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Firmicutes")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Epsilonbacteraeota")
plot_bar(gpsfb, "sampling_period", "Abundance")


```

## Leaves - Did the inoculation work?

Just LEAVES
```{r}

S5.leaf <- prune_samples(sample_data(S5)$tissue=="Leaf", S5)

#takes about 5 min on SF macbook pro
df.leaf <- otu_table(S5.leaf) %>% 
  as.data.frame()

df.meta.leaf <- sample_data(S5.leaf) %>% 
  data.frame()

#make generic for downstream analysis
df <- df.leaf
df.meta <- df.meta.leaf

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#for PERMANOVA with all chemistry
comm.df.16S <- comm.df

```

#### Plot PCA - Aitchison

Just LEAVES
```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

ggplot(data = df.PCA,
       aes(x = PC1,
           y = PC2,
           fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
 ggforce::geom_mark_ellipse(aes(color = orig_soil),
                           alpha = 0.2) +
  labs(title = NULL,
       caption = "Ellipses are visual guides and have no statistical meaning",
       shape = "Oil Addition",
       fill = "Inoculum",
       color = "Inoculum",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)")) +
   facet_grid(cols = vars(sampling_period),
             labeller = labeller(sampling_period = c("N16" = "6 Months", "J17" = "1 Year", "N17" = "1.5 Years", "J18" = "2 Years"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL)

ggsave("S5_16S_LEAVES_through_time.png", width = 7, height = 5.5)

```
## NMDS
```{r}

A <- comm.df
A <- metaMDS(comm = A, distance = "euclidean")

cbind(df.meta, A$points) %>% 
  ggplot(aes(x = MDS1,
             y = MDS2,
             fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
  # ggforce::geom_mark_ellipse(aes(linetype = plant_trt),
  #                            alpha = 0.2) +
  # scale_linetype_manual(labels = c("No Plant",
  #                                  "Planted"),
  #                       values = c("dashed", "solid")) +
  labs(title = NULL,
       caption = paste0("Stress = ", round(A$stress, 2)),
       shape = "Oil Addition",
       fill = "Inoculum",
       linetype = "Plant Treatment") +
    # facet_grid(cols = vars(sampling_period),
             # labeller = labeller(sampling_period = c("N16" = "6 Months", "J17" = "1 Year", "N17" = "1.5 Years", "J18" = "2 Years"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL)

```
## Alternative NMDS Plot

The panels I made above were too much and there were some critiques of the PCA, so I've made something that is visually simpler and is an NMDS.  It's hard to see the yellow though, so I'm going to change the colors to something darker.

```{r}


cbind(df.meta, A$points) %>%
  group_by(oil_added, orig_soil, sampling_period) %>% 
  dplyr::summarise(MDS1.mean = mean(MDS1),
            MDS2.mean = mean(MDS2)) %>% 
  ggplot(aes(x = MDS1.mean,
             y = MDS2.mean,
             group = interaction(oil_added, orig_soil))) +
  # geom_link2(aes(alpha = sampling_period),
  #            size = 3) +
  geom_path(aes(color = orig_soil),
            arrow = arrow(type = "open", 
                          length = unit(2, "mm")),
            size = 0.5,
            show.legend = TRUE) +
  geom_point(data = . %>% 
               filter(sampling_period=="N16"),
             aes(shape = oil_added,
                 fill = orig_soil),
             size = 3) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values = c(cPAL, "black", "gray")) +
  scale_color_manual(values = c(cPAL, "black", "gray")) +
  theme_bw() +
  guides(fill = guide_legend(override.aes=list(shape=22, size = 4, linetype = NULL)),
         color = guide_legend(override.aes=list(shape=22, size = 4, linetype = NULL)),
         shape = guide_legend(override.aes=list(shape=c(21,24), size = 4, linetype = NULL))) +
  labs(title = NULL,
       caption = paste0("Stress = ", round(A$stress, 2)),
       shape = "Oil Addition",
       color = "Inoculum",
       fill = "Inoculum",
       x = "MDS1",
       y = "MDS1") +
  theme(legend.position = "bottom" ,legend.direction = "vertical",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank())

ggsave("S5_16S_LEAVES_through_time_NMDS_vector.png", width = 8, height = 6)
```
## J18 Only to compare to Duke Results

```{r}

S5.leaf <- prune_samples(sample_data(S5)$tissue=="Leaf", S5)
S5.leaf.J18 <- prune_samples(sample_data(S5.leaf)$sampling_period=="J18", S5.leaf)

#takes about 5 min on SF macbook pro
df.leaf <- otu_table(S5.leaf.J18) %>% 
  as.data.frame()

df.meta.leaf <- sample_data(S5.leaf.J18) %>% 
  data.frame()

#make generic for downstream analysis
df <- df.leaf
df.meta <- df.meta.leaf

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#for PERMANOVA with all chemistry
comm.df.16S <- comm.df

## NMDS

A <- comm.df
A <- metaMDS(comm = A, distance = "euclidean")

cbind(df.meta, A$points) %>% 
  ggplot(aes(x = MDS1,
             y = MDS2,
             fill = orig_soil,
           shape = oil_added)) +
  my.scale_aes +
  # ggforce::geom_mark_ellipse(aes(linetype = plant_trt),
  #                            alpha = 0.2) +
  # scale_linetype_manual(labels = c("No Plant",
  #                                  "Planted"),
  #                       values = c("dashed", "solid")) +
  labs(title = NULL,
       caption = paste0("Stress = ", round(A$stress, 2)),
       shape = "Oil Addition",
       fill = "Inoculum",
       linetype = "Plant Treatment") +
    # facet_grid(cols = vars(sampling_period),
             # labeller = labeller(sampling_period = c("N16" = "6 Months", "J17" = "1 Year", "N17" = "1.5 Years", "J18" = "2 Years"))) +
  theme(legend.position = "bottom" ,legend.direction = "vertical") +
  scale_color_manual(values = cPAL)

set.seed(1)
mod1 <- adonis(comm.df ~ orig_soil*oil_added, 
               data = df.meta, 
               permutations = 9999, 
               method = "euclidean",
               strata = df.meta$sampling_period) 
mod1

```


## Permanova - Aitchison

Just Leaves
```{r}

set.seed(1)
mod1 <- adonis(comm.df ~ orig_soil*oil_added, data = df.meta, 
               permutations = 9999, 
               method = "euclidean",
               strata = df.meta$sampling_period) 
mod1

A <- vegdist(x = comm.df, method = "euclidean")
B <- betadisper(d = A, group = df.meta$orig_soil )

anova(B)

B <- betadisper(d = A, group = df.meta$oil_added )
anova(B)


set.seed(1)
mod1 <- dbrda(comm.df ~ orig_soil*oil_added + Condition(sampling_period),
              distance = "euclidean",
              data = df.meta)

anova(mod1, by = "margin", permutations = 9999)

```


Sampling_period + sampling_period:orig_soil are the only levels that are significant.  If you remove sampling period, nothing else is significant.  dbRDA isn't a good fit.



### What taxa are responsible?

```{r}

#quick approach based off a phyloseq tutorial
#https://joey711.github.io/phyloseq/preprocess.html

S5.leaf.2 <- S5.leaf

transform_sample_counts(S5.leaf.2, function(OTU) OTU/sum(OTU) )
#S5.roots = filter_taxa(S5.roots, function(x) sum(x > 3) > (0.2*length(x)), TRUE)

total = median(sample_sums(S5.leaf.2))
standf = function(x, t=total) round(t * (x / sum(x)))
gps = transform_sample_counts(S5.leaf.2, standf)

#Filter the taxa using a cutoff of 3.0 for the Coefficient of Variation
#gpsf = filter_taxa(gps, function(x) sd(x)/mean(x) > 3.0, TRUE)

#Subset the data to Proteobacteria
gpsfb = subset_taxa(gps, Phylum=="Proteobacteria")

#Now let’s summarize this slice of the data with some graphics.

plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Bacteroidetes")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Cyanobacteria")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Actinobacteria")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Firmicutes")
plot_bar(gpsfb, "sampling_period", "Abundance")

# Other phyla
gpsfb = subset_taxa(gps, Phylum=="Epsilonbacteraeota")
plot_bar(gpsfb, "sampling_period", "Abundance")


```


##  STOPPED HERE


```{r}
set.seed(1)

S5.ord <- ordinate(S5.roots, distance = "bray")

B <- plot_ordination(S5.roots, S5.ord, type="samples", color="season")

#What if you get rid of Cyanobactera?

S5.roots_noCB <- subset_taxa(S5.roots, Phylum!="Cyanobacteria")
S5.ord <- ordinate(S5.roots_noCB, distance = "bray")

B <- plot_ordination(S5.roots_noCB, S5.ord, type="samples", color="season")

#delve into it
S5.roots_noCB <- subset_taxa(S5.roots, Phylum=="Proteobacteria" & taxa_sums(S5.roots) > 100)
plot_bar(S5.roots_noCB, x = "oil_added", fill = "Phylum", facet_grid = ~ season)

```

## Plot NMDS - BC

Used PCoA so it can be compared to the PCA (which is the equivalent when PCoA is based on Euclidean)

```{r}
set.seed(1)

SF1.ord <- ordinate(SF1.post.16S, method = "NMDS", distance = "bray")

B <- plot_ordination(SF1.post.16S, SF1.ord, type="samples", color="oil")

ggplot(data = B$data,
       aes(x = NMDS1,
           y = NMDS2,
          shape = soil,
          fill = oil)) +
  my.scale_aes +
  labs(title = "SF1 16S NMDS on Bray-Curtis - post experiment", 
       caption = paste("stress = ", round(SF1.ord$stress, digits = 2), sep = " "))

```

## Permanova - BC

```{r}

#uses un-transformed OTU table
set.seed(1)
mod1 <- adonis(df ~ soil*oil, data = df.meta, permutations = 9999, method = "bray") 
mod1

A <- vegdist(x = df, method = "bray")
B <- betadisper(d = A, group = df.meta$soil )

anova(B)

```

Bray-Curtis may be significant for soil, but also is significant for beta dispersion.

### Plot and PERMANOVA (adonis) of 16S - Jaccard index

```{r}

#make OTUs binary

df[df > 0] <- 1

otu_table(SF1.post.16S) <- otu_table(df, taxa_are_rows = FALSE)

df.meta <- sample_data(SF1.post.16S) %>% 
  data.frame()

```

## Plot PCoA - Jaccard

Used PCoA so it can be compared to the PCA (which is the equivalent when PCoA is based on Euclidean)

```{r}
set.seed(1)

SF1.ord <- ordinate(SF1.post.16S, method = "PCoA", distance = "jaccard", binary = TRUE)

B <- plot_ordination(SF1.post.16S, SF1.ord, type="samples", color="oil")

ggplot(data = B$data,
       aes(x = Axis.1,
           y = Axis.2,
           shape = soil,
           fill = oil)) +
  my.scale_aes +
  labs(title = "PCoA on 16S Metagenome - Jaccard",
       x = "Axis 1",
       y = "Axis 2")
```

## Plot NMDS - Jaccard


```{r}
set.seed(1)

SF1.ord <- ordinate(SF1.post.16S, method = "NMDS", distance = "jaccard", binary = TRUE)

B <- plot_ordination(SF1.post.16S, SF1.ord, type="samples", color="oil")

ggplot(data = B$data,
       aes(x = NMDS1,
           y = NMDS2,
           shape = soil,
           fill = oil)) +
  my.scale_aes +
  labs(title = "SF1 16S NMDS on Jacard - post experiment")

```

## Permanova - Jaccard

```{r}

set.seed(1)
mod1 <- adonis(df ~ soil*oil, data = df.meta, permutations = 9999, method = "jaccard", binary = TRUE) 
mod1

A <- vegdist(x = df, method = "jaccard", binary = TRUE)
B <- betadisper(d = A, group = df.meta$soil )

anova(B)

```

Same as Bray-Curtis, significant for soil but also for dispersion.
## ITS

### Plot and PERMANOVA (adonis) of ITS - Aitchison

```{r}

SF1.post.ITS <- prune_samples(sample_data(SF1.ITS)$time=="Post-Experiment", SF1.ITS)

df <- otu_table(SF1.post.ITS) %>% 
  as.data.frame()

df.meta <- sample_data(SF1.post.ITS) %>% 
  data.frame()

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#For analysis with chemistry below
comm.df.ITS <- comm.df

```

## Permanova - Aitchison

```{r}

set.seed(1)
mod1 <- adonis(comm.df ~ soil*oil, data = df.meta, permutations = 9999, method = "euclidean") 
mod1

```
Not significant, no need for dispersion test.

## Plot PCA - Aitchison

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

PCA.ITS <- ggplot(data = df.PCA, 
                aes(x = PC1, 
                    y = PC2, 
                    shape = soil,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCA on ITS Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)"))

PCA.ITS
```


### PERMANOVA (adonis) of ITS - Bray-Curtis

## Plot

Used PCoA so it can be compared to the PCA (which is the equivalent when PCoA is based on Euclidean)

```{r}
set.seed(1)

SF1.ord <- ordinate(SF1.post.ITS, method = "PCoA", distance = "bray")

B <- plot_ordination(SF1.post.ITS, SF1.ord, type="samples", color="oil")

ggplot(data = B$data,
       aes(x = Axis.1,
           y = Axis.2,
           shape = soil,
           fill = oil)) +
  my.scale_aes +
  labs(title = "PCoA on ITS Metagenome",
       x = "Axis 1",
       y = "Axis 2")

```
## Plot

Used PCoA so it can be compared to the PCA (which is the equivalent when PCoA is based on Euclidean)

```{r}
set.seed(1)

SF1.ord <- ordinate(SF1.post.ITS, method = "NMDS", distance = "bray")

B <- plot_ordination(SF1.post.ITS, SF1.ord, type="samples", color="oil")

ggplot(data = B$data,
       aes(x = NMDS1,
           y = NMDS2,
           shape = soil,
           fill = oil)) +
  my.scale_aes +
  labs(title = "SF1 ITS NMDS on Bray-Curtis - post experiment", 
       caption = paste("stress = ", round(SF1.ord$stress, digits = 2), sep = " "))

```

## Permanova
```{r}

set.seed(1)
mod1 <- adonis(df ~ soil*oil, data = df.meta, permutations = 9999, method = "bray") 
mod1

```
Not signifcant, not need for dispersion test.

### PERMANOVA (adonis) of ITS - Jaccard index

```{r}

df[df > 0] <- 1

otu_table(SF1.post.ITS) <- otu_table(df, taxa_are_rows = FALSE)

df.meta <- sample_data(SF1.post.ITS) %>% 
  data.frame()

```
## Plot

Used PCoA so it can be compared to the PCA (which is the equivalent when PCoA is based on Euclidean)

```{r}
set.seed(1)

SF1.ord <- ordinate(SF1.post.ITS, method = "PCoA", distance = "jaccard", binary = TRUE)

B <- plot_ordination(SF1.post.ITS, SF1.ord, type="samples", color="oil")

ggplot(data = B$data,
       aes(x = Axis.1,
           y = Axis.2,
           shape = soil,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCoA on ITS Metagenome - Jaccard",
       x = "Axis 1",
       y = "Axis 2")
```
## Plot

Used PCoA so it can be compared to the PCA (which is the equivalent when PCoA is based on Euclidean)

```{r}
set.seed(1)

SF1.ord <- ordinate(SF1.post.ITS, method = "NMDS", distance = "jaccard", binary = TRUE)

B <- plot_ordination(SF1.post.ITS, SF1.ord, type="samples", color="oil")

ggplot(data = B$data,
       aes(x = NMDS1,
           y = NMDS2,
           shape = soil,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "SF1 ITS NMDS on Jacard - post experiment")

```

## Permanova
```{r}

set.seed(1)
mod1 <- adonis(df ~ soil*oil, data = df.meta, permutations = 9999, method = "jaccard", binary = TRUE) 
mod1

A <- vegdist(x = df, method = "jaccard", binary = TRUE)
B <- betadisper(d = A, group = df.meta$soil )

anova(B)
```

Significant for soil and for beta dispersion.

## Plot Figture of ITS and 16S PCA

This is based on Aitchison distance.

```{r}

top <- plot_grid(PCA.ITS + 
            theme(legend.position = "none") +
            labs(title = NULL, 
                 subtitle = "Fungi"),
            PCA.16S + 
            theme(legend.position = "none") + 
            labs(title = NULL,
                 subtitle = "Bacteria"),
          labels = "AUTO")

bottom <- get_legend(PCA.16S + 
            theme(legend.position = "bottom"))

final <- plot_grid(top, bottom, ncol = 1, rel_heights = c(1, 0.1))

final 

ggsave(here::here("figs", "manuscript", "beta_div", "SF1_beta_div.png"), width = 10, height = 7, units = "in")
```


## RDA model with chemistry 

To look at how chemistry explains species composition.  First CLR transform chemistry.  

Note that vegan uses QR composition so scaling and centering isn't necessary:
https://stat.ethz.ch/pipermail/r-sig-ecology/2010-April/001196.html


```{r}
#Scale chemistry

SF1.post.ITS <- prune_samples(sample_data(SF1.ITS)$time=="Post-Experiment", SF1.ITS)


df.meta <- sample_data(SF1.post.ITS) %>% 
  data.frame()

#Convert %C and N to ppm
df.meta$Percent.Carbon <- df.meta$Percent.Carbon*10000
df.meta$Percent.Nitrogen <- df.meta$Percent.Nitrogen*10000

clr.df.meta <- clr(df.meta[,c(5:7,9:15)])

clr.df.meta <- as.data.frame(clr.df.meta)
clr.df.meta$pH <- df.meta$pH
clr.df.meta$soil <- df.meta$soil
clr.df.meta$oil <- df.meta$oil
```

### Test for collinearity

Test if we can improve the model by removing chemical measurements with high collinearity.
```{r}

#make full model without treatments
RDA.16S <- rda(comm.df.16S ~ Calcium + Copper + Magnesium + pH + Phosphorus + Potassium + Sodium + Sulfur + Zinc + Percent.Carbon + Percent.Nitrogen, data = clr.df.meta)

#Get VIF scores
vif.cca(RDA.16S)
```

Returns NA for some variables and the rest are extremely high, indicating multicollinearity. Doesn't necessarily mean any should be thrown out, but the NAs mean that some are perfectly correlated with others which holds up in the correlation plot below.

```{r}

#how bad are the correlations among explanatory factors?
ggcorrplot(cor(clr.df.meta[,1:11]), hc.order = TRUE, type = "lower",
           lab = TRUE)

```

Remove variables to improve VIF scores by removing one of any paris with a correlation of 0.9 or higher.  So I'm only trying to remove positivie relationships.

```{r}

#make full model without treatments
RDA.16S <- rda(comm.df.16S ~ Calcium + Magnesium + pH + Phosphorus + Potassium + Sulfur + Percent.Nitrogen, data = clr.df.meta)

#Copper is highly correlated with sulfur and % Carbon
#Zinc is highly correlated with Calcium
#Sodium is highly correlated with potassium
#pH and percent Carbon are highly correlated.  Since % Carbon is a no-brainer with the oil addition, I'm going to leave pH in since it provides more information about how the chemnistry changes.

#Get VIF scores
vif.cca(RDA.16S)

```

#### 16S Model
```{r}

#Check that the best model are the treatments as explanatory variables.  This would make sense because all the chemistry should have more variation than a categorical variable.

RDA.16S <- rda(comm.df.16S ~ soil + oil + Calcium + Copper + Magnesium + pH + Phosphorus + Potassium + Sodium + Sulfur + Zinc + Percent.Carbon + Percent.Nitrogen, data = clr.df.meta)

null.RDA.16S <- rda(comm.df.16S ~ 1, data = clr.df.meta)

ordistep(null.RDA.16S, scope = formula(RDA.16S), direction = "forward")

```

The above chooses soil only.  Let's see what happens if we take out soil and oil and use the elements selected above.

```{r}


RDA.16S <- rda(comm.df.16S ~ Calcium + Magnesium + pH + Phosphorus + Potassium + Sulfur + Percent.Nitrogen, data = clr.df.meta)

null.RDA.16S <- rda(comm.df.16S ~ 1, data = clr.df.meta)

ordistep(null.RDA.16S, scope = formula(RDA.16S), direction = "both")

```

Independently, Potassium, Sodium, Sulfur, and Magnesium are signficiant.  But Potassium is the only element that is significant when the whole model is run.

#### ITS Model
```{r}

RDA.ITS <- rda(comm.df.ITS ~ soil + oil + Calcium + Copper + Magnesium + pH + Phosphorus + Potassium + Sodium + Sulfur + Zinc + Percent.Carbon + Percent.Nitrogen, data = clr.df.meta)

null.RDA.ITS <- rda(comm.df.ITS ~ 1, data = clr.df.meta)

ordistep(null.RDA.ITS, scope = formula(RDA.ITS), direction = "both")

```

The above chooses the null model.  Let's see what happens if we take out soil and oil.

```{r}

RDA.ITS <- rda(comm.df.ITS ~ Calcium + Copper + Magnesium + pH + Phosphorus + Potassium + Sodium + Sulfur + Zinc + Percent.Carbon + Percent.Nitrogen, data = clr.df.meta)

null.RDA.ITS <- rda(comm.df.ITS ~ 1, data = clr.df.meta)

ordistep(null.RDA.ITS, scope = formula(RDA.ITS), direction = "forward")

```

Null is still the best model.

##  What about Time? i.e. pre/post differences?

Also, check for contamination.

#### Plot and PERMANOVA (adonis) of 16S - Aitchison

```{r}

df <- otu_table(SF1.16S) %>% 
  as.data.frame()

df.meta <- sample_data(SF1.16S) %>% 
  data.frame()

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

```

#### Plot PCA - Aitchison

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

PCA.16S.all <- ggplot(data = df.PCA, 
                aes(x = PC1, 
                    y = PC2,
                    color = time,
                    shape = soil,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCA on 16S Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)"))

PCA.16S.all

```

#### Plot and PERMANOVA (adonis) of ITS - Aitchison

```{r}

df <- otu_table(SF1.ITS) %>% 
  as.data.frame()

df.meta <- sample_data(SF1.ITS) %>% 
  data.frame()

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

```

#### Plot PCA - Aitchison

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

PCA.ITS.all <- ggplot(data = df.PCA, 
                aes(x = PC1, 
                    y = PC2,
                    color = time,
                    shape = soil,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCA on ITS Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)"))

PCA.ITS.all

```

##  What about within autoclave treamtents?  Can you tell the difference between oiled and unoiled samples?


#### Plot and PERMANOVA (adonis) of 16S - Aitchison

```{r}

SF1.live.16S <- prune_samples(sample_data(SF1.16S)$soil=="Live Soil" & sample_data(SF1.16S)$time=="Post-Experiment", SF1.16S)

df <- otu_table(SF1.live.16S) %>% 
  as.data.frame()

df.meta <- sample_data(SF1.live.16S) %>% 
  data.frame()

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#Convert %C and N to ppm
df.meta$Percent.Carbon <- df.meta$Percent.Carbon*10000
df.meta$Percent.Nitrogen <- df.meta$Percent.Nitrogen*10000

clr.df.meta <- clr(df.meta[,c(5:7,9:15)])

clr.df.meta <- as.data.frame(clr.df.meta)
clr.df.meta$pH <- df.meta$pH
clr.df.meta$soil <- df.meta$soil
clr.df.meta$oil <- df.meta$oil

```

#### Plot PCA - Aitchison

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

PCA.16S.live <- ggplot(data = df.PCA, 
                aes(x = PC1, 
                    y = PC2,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCA on 16S Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)"))

PCA.16S.live

```
#### PERMANOVA

```{r}

adonis(comm.df ~ oil, method = "euclidean", permutations = 9999, data = df.meta)

RDA.16S <- rda(comm.df ~ Calcium + Copper + pH + Phosphorus  + Zinc + Percent.Carbon + Percent.Nitrogen + Potassium + Sodium + Sulfur + Magnesium, data = clr.df.meta, )

null.RDA.16S <- rda(comm.df ~ 1, data = clr.df.meta)

ordistep(null.RDA.16S, scope = formula(RDA.16S), direction = "both")

```

#### Plot and PERMANOVA (adonis) of 16S - Aitchison

```{r}

SF1.auto.16S <- prune_samples(sample_data(SF1.16S)$soil=="Autoclaved Soil" & sample_data(SF1.16S)$time=="Post-Experiment", SF1.16S)

df <- otu_table(SF1.auto.16S) %>% 
  as.data.frame()

df.meta <- sample_data(SF1.auto.16S) %>% 
  data.frame()

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

#Convert %C and N to ppm
df.meta$Percent.Carbon <- df.meta$Percent.Carbon*10000
df.meta$Percent.Nitrogen <- df.meta$Percent.Nitrogen*10000

clr.df.meta <- clr(df.meta[,c(5:7,9:15)])

clr.df.meta <- as.data.frame(clr.df.meta)
clr.df.meta$pH <- df.meta$pH
clr.df.meta$soil <- df.meta$soil
clr.df.meta$oil <- df.meta$oil
```

#### Plot PCA - Aitchison

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

PCA.16S.auto <- ggplot(data = df.PCA, 
                aes(x = PC1, 
                    y = PC2,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCA on 16S Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)"))

PCA.16S.auto

```
#### PERMANOVA

```{r}

adonis(comm.df ~ oil, method = "euclidean", permutations = 9999, data = df.meta)

RDA.16S <- rda(comm.df ~ Calcium + Copper + pH + Phosphorus  + Zinc + Percent.Carbon + Percent.Nitrogen + Potassium + Sodium + Sulfur + Magnesium, data = clr.df.meta, )

null.RDA.16S <- rda(comm.df ~ 1, data = clr.df.meta)

ordistep(null.RDA.16S, scope = formula(RDA.16S), direction = "both")
```

### ITS

#### Plot and PERMANOVA (adonis) of ITS - Aitchison

```{r}

SF1.live.ITS <- prune_samples(sample_data(SF1.ITS)$soil=="Live Soil" & sample_data(SF1.ITS)$time=="Post-Experiment", SF1.ITS)

df <- otu_table(SF1.live.ITS) %>% 
  as.data.frame()

df.meta <- sample_data(SF1.live.ITS) %>% 
  data.frame()

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

```

#### Plot PCA - Aitchison

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

PCA.ITS.live <- ggplot(data = df.PCA, 
                aes(x = PC1, 
                    y = PC2,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCA on ITS Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)"))

PCA.ITS.live

```
#### PERMANOVA

```{r}

adonis(comm.df ~ oil, method = "euclidean", permutations = 9999, data = df.meta)

```

#### Plot and PERMANOVA (adonis) of ITS - Aitchison

```{r}

SF1.auto.ITS <- prune_samples(sample_data(SF1.ITS)$soil=="Autoclaved Soil" & sample_data(SF1.ITS)$time=="Post-Experiment", SF1.ITS)

df <- otu_table(SF1.auto.ITS) %>% 
  as.data.frame()

df.meta <- sample_data(SF1.auto.ITS) %>% 
  data.frame()

clr.df <- clr(df)
pca = prcomp(clr.df, center = FALSE, scale. = FALSE)

PCA.loadings=pca$rotation
PCA.scores = pca$x

#For below analyses
comm.df <- clr.df %>% 
  as.data.frame()

```

#### Plot PCA - Aitchison

```{r}

#get summary data
pca.sum <- as.data.frame(summary(pca)$importance)
df.PCA <- cbind(PCA.scores,df.meta)

PCA.ITS.auto <- ggplot(data = df.PCA, 
                aes(x = PC1, 
                    y = PC2,
                    fill = oil)) +
  my.scale_aes +
  labs(title = "PCA on ITS Metagenome - Aitchison",
       x = paste("PC1 (", round(pca.sum$PC1[2]*100, digits = 1), "%)"),
       y = paste("PC2 (", round(pca.sum$PC2[2]*100, digits = 1), "%)"))

PCA.ITS.auto

```
#### PERMANOVA

```{r}

adonis(comm.df ~ oil, method = "euclidean", permutations = 9999, data = df.meta)

```